nemo/export/trt_llm/nemo/sentencepiece_tokenizer.py

```diff
@@ -0,0 +1,249 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from typing import Dict, List, Optional, Union
+
+import numpy as np
+import sentencepiece
+import torch
+
+
+class SentencePieceTokenizer:
```

Comment:
Why is NeMo's SPE Tokenizer being copy pasted here? Why not import it?

Reply:
I think its because export relies on huggingface style tokenizers so the API is slightly changed compared to NeMo SPE. I have a function to monkey patch the tokenizer maybe we could use that?
[https://github.com/NVIDIA/NeMo/pull/8743/files#diff-5577401004be5ef0698866e5cb14f7a224547e87de1286746d731025fdb4018fR580](url)

Reply:
If we can move to use NeMo tokenizer without too much effort, let's do that. If it's too much effort, this duplication is fine but it can diverge due to modifications of NeMo tokenizer (potentially)

Reply:
In addition to Jimmy, we need to keep this part self contained due to the NIM requirements unfortunately.

Reply:
Unresolving this as we should figure out how to remove the duplicate implementations. We can merge this PR as is but we should send a follow up PR fixing this.

Reply:
I agree but coming up with a solution will be a bit hard since we won't be able to import any thing from the other modules of the nemo.